omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/LocationResource1_8.java

```diff
@@ -249,39 +251,54 @@ public void purge(Location location, RequestContext context) throws ResponseExce
 	protected NeedsPaging<Location> doGetAll(RequestContext context) {
 		return new NeedsPaging<Location>(Context.getLocationService().getAllLocations(context.getIncludeAll()), context);
 	}
-	
+
 	/**
 	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doSearch(org.openmrs.module.webservices.rest.web.RequestContext)
-	 *      A query string and/or a tag uuid can be passed in; if both are passed in, returns an
+	 *      A query string and/or a tag (referenced by name or uuid) can be passed in; if both are passed in, returns an
 	 *      intersection of the results; excludes retired locations
 	 */
 	@Override
 	protected PageableResult doSearch(RequestContext context) {
-		
-		LocationService locationService = Context.getLocationService();
-		
-		String tagUuid = context.getParameter("tag");
+
+		String tag = context.getParameter("tag");
 		String query = context.getParameter("q");
-		
-		List<Location> locationsByTag = null;
-		List<Location> locationsByQuery = null;
-		
-		if (tagUuid != null) {
-			LocationTag locationTag = locationService.getLocationTagByUuid(tagUuid);
-			locationsByTag = locationService.getLocationsHavingAllTags(Arrays.asList(locationTag));
+
+		List<Location> locationsByTag = Collections.emptyList();
+		List<Location> locationsByQuery = Collections.emptyList();
+
+		if (tag != null) {
+			try {
+				Context.addProxyPrivilege(VIEW_LOCATIONS); //Not using PrivilegeConstants.VIEW_LOCATIONS which was removed in platform 1.11+
+				Context.addProxyPrivilege("Get Locations"); //1.11+
+
+				LocationTag locationTag = Context.getLocationService().getLocationTagByUuid(tag);
+				if (locationTag == null) {
+					locationTag = Context.getLocationService().getLocationTagByName(tag);
+				}
+
+				if (locationTag != null) {
+					locationsByTag = Context.getLocationService().getLocationsHavingAllTags(Arrays.asList(locationTag));
+				}
+			} finally {
+				Context.removeProxyPrivilege(VIEW_LOCATIONS); //Not using PrivilegeConstants.VIEW_LOCATIONS which was removed in platform 1.11+
+				Context.removeProxyPrivilege("Get Locations"); //1.11+
+			}
 		}
-		
+
 		if (query != null) {
-			locationsByQuery = locationService.getLocations(query);
+			locationsByQuery = Context.getLocationService().getLocations(query);
 		}
-		
-		if (locationsByTag == null) {
+
+		// note that here we check if the actual query parameters are null, not the results
+		// because, for instance, if a tag parameter is specified and no locations are found with that tag, we want to return an empty list,
+		// regardless of the query parameter, and vice versa; note the default value of the results are set to empty lists above
```

Comment:
Get rid of this entire comment, after changes made above.  Anything that is not null should be part of the intersection of results.